tls_for_test.go

```diff
@@ -0,0 +1,220 @@
+package main_test
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"net"
+	"os"
+	"testing"
+	"time"
+)
+
+type tlsHelpers struct {
+	ca                *x509.Certificate
+	caPrivKey         *ecdsa.PrivateKey
+	caPEM             *bytes.Buffer
+	caPrivKeyPEM      *bytes.Buffer
+	caFile            string
+	caPrivKeyFile     string
+	serverCert        *x509.Certificate
+	serverPrivKey     *ecdsa.PrivateKey
+	serverPEM         *bytes.Buffer
+	serverFile        string
+	serverPrivKeyPEM  *bytes.Buffer
+	serverPrivKeyFile string
+	serverCertPair    tls.Certificate
+	certpool          *x509.CertPool
+	serverTLSConf     *tls.Config
+	clientCert        *x509.Certificate
+	clientPrivKey     *ecdsa.PrivateKey
+	clientPEM         *bytes.Buffer
+	clientFile        string
+	clientPrivKeyPEM  *bytes.Buffer
+	clientPrivKeyFile string
+	clientTLSConf     *tls.Config
+}
```

Comment:
Now that this is up and running, most of these fields can probably be dropped in favor of local vars in the function. Only a few of these get used outside of this file.